applications/serial_lte_modem/src/twi/slm_at_twi.c

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <logging/log.h>
+#include <zephyr.h>
+#include <drivers/i2c.h>
+#include <stdio.h>
+#include "slm_util.h"
+#include "slm_at_twi.h"
+
+LOG_MODULE_REGISTER(slm_twi, CONFIG_SLM_LOG_LEVEL);
+
+#define TWI_MAX_INSTANCE	4
```

Comment:
DTS-based?

Reply:
Now the max instance is determined by according to CONFIG_HAS_HW_NRF_TWIM3/CONFIG_HAS_HW_NRF_TWIM2/CONFIG_HAS_HW_NRF_TWIM1/CONFIG_HAS_HW_NRF_TWIM0

If TWI(i2c) instance is enabled in device tree, the instance will be bind at initialization phase.